src/main/java/org/glassfish/enterprise/concurrent/ManagedThreadFactoryImpl.java

```diff
@@ -43,15 +45,18 @@ public class ManagedThreadFactoryImpl implements ManagedThreadFactory {
     private Lock lock; // protects threads and stopped
 
     private String name;
-    final private ContextSetupProvider contextSetupProvider;
+    final protected ContextSetupProvider contextSetupProvider;
```

Comment:
Maybe provide a getter here instead of making private into protected?

Reply:
Getter would make it a part of ManagedThreadFactoryImpl's public API, which I didn't want to.

The purpose of making it protected was to enable storing context during loading from jndi lookup, which I did to pass TCK. It was later removed (https://github.com/payara/Payara/commit/44275aa105186aa4aaa49e96a446feffb50bcfc2, the test is going to be challenged), but some implementation may decide to use it. The accompanying field is `savedContextHandleForSetup`.